code/modules/antagonists/heretic/heretic_monster.dm

```diff
@@ -0,0 +1,45 @@
+///Tracking reasons
+/datum/antagonist/heretic_monster
+	name = "\improper Eldritch Horror"
+	roundend_category = "Heretics"
+	job_rank = ROLE_HERETIC
+	antag_hud_type = ANTAG_HUD_HERETIC_BEAST
+	antag_hud_name = "heretic_beast"
+	/// Our master (a heretic)'s mind.
+	var/datum/mind/master
+
+/datum/antagonist/heretic_monster/farewell()
+	if(master?.current)
+		to_chat(master.current, "<span class='warning'>The essence of [owner], your servant, fades from your mind.</span>")
+	if(owner.current)
+		to_chat(owner.current, "<span class='deconversion_message'>Your mind begins to fill with haze - your master is no longer[master ? " [master]":""], you are free!</span>")
+		owner.current.visible_message("<span class='deconversion_message'>[owner.current] looks like [owner.current.p_theyve()] been freed from the chains of the Mansus!</span>")
+
+	master = null
+	return ..()
+
+/datum/antagonist/heretic_monster/add_owner_to_gamemode()
+	SSticker.mode.heretics += owner
+
+/datum/antagonist/heretic_monster/remove_owner_from_gamemode()
+	SSticker.mode.heretics -= owner
+/*
+ * Set our [master] var to a new mind.
+ */
+/datum/antagonist/heretic_monster/proc/set_owner(datum/mind/master)
+	src.master = master
+	//owner.enslave_mind_to_creator(master.current)
```

Comment:
Commented out?

Reply:
Proc doesnt exist here, need to come back to it to make mindslaves clearer